groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1431,10708 @@ void Optional_DataImp<TYPE>::emplace(
+        return UnspecifiedBoolUtil::makeValue(has_value());
+    }
+#endif  //#ifdef BSLS_COMPILERFEATURES_SUPPORT_OPERATOR_EXPLICIT
+};
+
+                            // ====================
+                            // class optional<TYPE>
+                            // ====================
+
+#ifdef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+template <class TYPE>
+class optional<TYPE, false> : public std::optional<TYPE> {
+    typedef std::optional<TYPE> optionalBase;
+
+  public:
+    using optionalBase::optionalBase;
+
+    optional(const optional& rhs);  // IMPLICIT
+        // If 'rhs' contains a value, initialize the 'value_type' object using
+        // '*rhs'. Otherwise, create a disengaged 'optional'.  Needed because
+        // we declare a move assignment operator.
+
+    optional(optional&& rhs);  // IMPLICIT
+        // If 'rhs' contains a value, initialize the 'value_type' object by
+        // move construction from '*rhs'. Otherwise, create a disengaged
+        // 'optional'. 'rhs' is left in a valid, but unspecified state.
+
+    template <class ANY_TYPE = TYPE>
+    optional(const std::optional<ANY_TYPE>& rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_STD_OPTIONAL_LVAL
+        BSLSTL_OPTIONAL_ENABLE_IF_NOT_EXPLICIT_CONSTRUCT(TYPE,ANY_TYPE))
+    // If specified 'rhs' contains a value, initialize the 'value_type'
+    // object by moving from '*rhs'.  Otherwise, create a disengaged
+    // 'optional'.  Use the currently installed default allocator to supply
+    // memory for this and any future 'value_type' objects.
+    : optionalBase(rhs)
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+    }
+
+    template <class ANY_TYPE = TYPE>
+    explicit optional(const std::optional<ANY_TYPE>& rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_STD_OPTIONAL_LVAL
+        BSLSTL_OPTIONAL_ENABLE_IF_EXPLICIT_CONSTRUCT(TYPE, ANY_TYPE))
+    // If specified 'rhs' contains a value, initialize the 'value_type'
+    // object by moving from '*rhs'.  Otherwise, create a disengaged
+    // 'optional'.  Use the currently installed default allocator to supply
+    // memory for this and any future 'value_type' objects.
+    : optionalBase(rhs)
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+    }
+
+    template <class ANY_TYPE = TYPE>
+    optional(std::optional<ANY_TYPE>&& rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_STD_OPTIONAL_RVAL
+        BSLSTL_OPTIONAL_ENABLE_IF_NOT_EXPLICIT_CONSTRUCT(TYPE, ANY_TYPE))
+    // If specified 'rhs' contains a value, initialize the 'value_type'
+    // object by moving from '*rhs'.  Otherwise, create a disengaged
+    // 'optional'.  Use the currently installed default allocator to supply
+    // memory for this and any future 'value_type' objects.
+    : optionalBase(std::forward<std::optional<ANY_TYPE> >(rhs))
```

Comment:
This is an rvalue reference, not forwarding reference, right? 
```suggestion
    : optionalBase(std::move(rhs))
```

Same thing below.

Reply:
of course, I have no idea where that pointless forward came from...  leaving this open until i check if tests need to be improved.

Reply:
I mean, the result is the same (rvalue ref in both cases), so there's not much to test here.

Reply:
ah, yes, true.